Core/KSPManager.cs

```diff
@@ -151,6 +152,103 @@ public KSP AddInstance(KSP ksp_instance)
+        /// <param name="new_path">The path where the new instance should be located.</param>
+        public void CloneInstance (KSP existing_instance, string new_name, string new_path)
+        {
+            if (!existing_instance.Valid)
+            {
+                throw new NotKSPDirKraken(existing_instance.GameDir(), "The specified instance is not a valid KSP instance.");
+            }
+
+            try
+            {
+                log.Debug("Copying directory.");
+                Utilities.CopyDirectory(existing_instance.GameDir(), new_path, true);
+
+                // Add the new instance to the registry
+                KSP new_instance = new KSP(new_path, new_name, User);
+                AddInstance(new_instance);
+            }
+            // Thrown by AddInstance() if created instance is not valid.
+            // Thrown f.e. if something went wrong with copying.
+            catch (NotKSPDirKraken kraken)
+            {
+                throw kraken;
+            }
+            // Thrown by CopyDirectory() if directory doesn't exist. Shouldn't be thrown anytime.
+            catch (DirectoryNotFoundKraken kraken)
+            {
+                throw kraken;
+            }
+            // Thrown by CopyDirectory() if the specified folder already exists and is not empty.
+            catch (IOException e)
+            {
+                throw e;
+            }
+        }
+
+        /// <summary>
+        /// Create a new fake KSP instance
+        /// </summary>
+        /// <param name="new_name">The name for the new instance.</param>
+        /// <param name="new_path">The loaction of the new instance.</param>
+        /// <param name="version">The version of the new instance. Should have a build number.</param>
+        /// <param name="DLC">Whether to fake the DLC too.</param>
+        /// <param name="dlcVersion">The version of the DLC. Can be null if DLC == false.</param>
+        public void FakeInstance(string new_name, string new_path, KspVersion version, bool DLC, string dlcVersion)
+        {
+            if (!version.IsValid())
+            {
+                throw new ArgumentOutOfRangeException(nameof(version), "The specified KSP version is not a valid version.");
+            }
+            if (Directory.Exists(new_path) && (Directory.GetFiles(new_path).Length != 0 || Directory.GetDirectories(new_path).Length != 0))
+            {
+                throw new BadInstallLocationKraken("The specified folder already exists and is not empty.");
+            }
+
+            try
+            {
+                log.DebugFormat("Creating folder structure and text files at {0} for KSP version {1}", Path.GetFullPath(new_path), version.ToString());
+
+                // Create a KSP root directory, containing a GameData folder, a buildID.txt/buildID64.txt and a readme.txt
+                Directory.CreateDirectory(new_path);
+                Directory.CreateDirectory(Path.Combine(new_path, "GameData"));
```

Comment:
Could this could create the other folders that the validation scripts currently do? This would help with being able to install mods.

https://github.com/KSP-CKAN/xKAN-meta_testing/blob/36ccd470cde1c6a02a8a3fc267e2eddad2b6640c/NetKAN/build.sh#L54-L59